python/lsst/pipe/tasks/ingestCalibs.py

```diff
@@ -0,0 +1,164 @@
+    def __init__(self, *args, **kwargs):
+        ArgumentParser.__init__(self, *args, **kwargs)
+        self.add_argument("-n", "--dry-run", dest="dryrun", action="store_true", default=False, help="Don't perform any action?")
+        self.add_argument("--create", action="store_true", help="Create new registry?")
+        self.add_argument("--validity", type=int, help="Calibration validity period (days)")  
+        self.add_argument("files", nargs="+", help="Names of file")
+
+class IngestCalibsConfig(Config):
+    """Configuration for IngestCalibsTask"""
+    parse = ConfigurableField(target=CalibsParseTask, doc="File parsing")
+    register = ConfigurableField(target=CalibsRegisterTask, doc="Registry entry")
+
+class IngestCalibsTask(Task):
+    """Task that generates registry for calibration images"""
+    ConfigClass = IngestCalibsConfig
+    ArgumentParser = IngestCalibsArgumentParser
+    _DefaultName = "ingestCalibs"
+
+    def __init__(self, *args, **kwargs):
+        Task.__init__(self, *args, **kwargs)
+        self.makeSubtask("parse")
+        self.makeSubtask("register")
+
+    @classmethod
+    def parseAndRun(cls):
+        """Parse the command-line arguments and run the Task"""
+        config = cls.ConfigClass()
+        parser = cls.ArgumentParser("ingestCalibs")
+        args = parser.parse_args(config)
+        task = cls(config=args.config)
+        task.run(args)
+
+    def run(self, args):
+        """Ingest all specified files and add them to the registry"""
+        with self.register.openRegistry(args.butler, create=args.create, dryrun=args.dryrun) as registry:
+            for infile in args.files:
+                fileInfo, hduInfoList = self.parse.getInfo(infile)
+                for info in hduInfoList:
+                    if info["obstype"] in self.register.config.tables:
+                        table = info["obstype"]
+                        info['path'] = infile
+                        if args.validity is not None:
+                            try: 
+                                info['validStart'] = (_convertToDate(info['calibDate'])
+                                                      - datetime.timedelta(args.validity)).isoformat()
+                                info['validEnd'] = (_convertToDate(info['calibDate'])
+                                                    + datetime.timedelta(args.validity)).isoformat()
+                            except TypeError:
+                                self.log.warn("Skipped setting validity period of %s" % args.validity)
+                        self.register.addRow(registry, info, dryrun=args.dryrun, create=args.create, table=table)
+                    else:
+                        self.log.warn("Skipped adding %s of obstype %s to registry" % (infile, info["obstype"]))
+            if args.dryrun:
+                print "Would update validity ranges to resolve overlaps"
```

Comment:
This should be a print function (add `from __future__ import print_function` at top of file), or an `info` log message.